src/pg_query_summary.c

```diff
@@ -0,0 +1,761 @@
+			add_from_clause_items(stmt->fromClause, CONTEXT_SELECT, state);
+			break;
+		}
+
+		case T_DeleteStmt: {
+			DeleteStmt *stmt = castNode(DeleteStmt, node);
+			add_from_clause_item(stmt->relation, CONTEXT_DML, state);
+			add_from_clause_items(stmt->usingClause, CONTEXT_SELECT, state);
+			break;
+		}
+
+		case T_CopyStmt: {
+			CopyStmt* stmt = castNode(CopyStmt, node);
+			add_from_clause_item(stmt->relation, CONTEXT_SELECT, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_AlterTableStmt: {
+			AlterTableStmt* stmt = castNode(AlterTableStmt, node);
+
+			// `ALTER INDEX index_name` is ignored.
+			if (stmt->objtype != OBJECT_INDEX)
+				add_from_clause_item(stmt->relation, CONTEXT_DDL, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateStmt:
+			add_from_clause_item(castNode(CreateStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_CreateTableAsStmt: {
+			CreateTableAsStmt* stmt = castNode(CreateTableAsStmt, node);
+			if (stmt->into && stmt->into->rel)
+				add_from_clause_item(stmt->into->rel, CONTEXT_DDL, state);
+
+			if (stmt->query)
+				pg_query_summary_walk_impl(stmt->query, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_TruncateStmt:
+			add_from_clause_items(castNode(TruncateStmt, node)->relations, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_ViewStmt: {
+			ViewStmt* stmt = castNode(ViewStmt, node);
+			add_from_clause_item(stmt->view, CONTEXT_DDL, state);
+			pg_query_summary_walk_impl(stmt->query, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_IndexStmt: {
+			IndexStmt* stmt = castNode(IndexStmt, node);
+
+			add_from_clause_item(stmt->relation, CONTEXT_DDL, state);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->indexParams) {
+				IndexElem* index_elem = lfirst(lc);
+				if (index_elem->expr)
+					pg_query_summary_walk_impl(index_elem->expr, state);
+			}
+
+			if (stmt->whereClause)
+				pg_query_summary_walk_impl(stmt->whereClause, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateTrigStmt:
+			add_from_clause_item(castNode(CreateTrigStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_RuleStmt: {
+			add_from_clause_item(castNode(RuleStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_VacuumStmt: {
+			VacuumStmt* stmt = castNode(VacuumStmt, node);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->rels) {
+				VacuumRelation* rel = lfirst(lc);
+				add_from_clause_item(rel->relation, CONTEXT_DDL, state);
+			}
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_RefreshMatViewStmt:
+			add_from_clause_item(castNode(RefreshMatViewStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_FuncCall:
```

Comment:
I would move this down right before `T_A_Expr` for consistency (since this is not part of the original tree walk in the Ruby implementation)